pkg/apis/scaler/v1alpha1/conversion_demand.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/palantir/k8s-spark-scheduler-lib/pkg/apis/scaler/v1alpha2"
+	werror "github.com/palantir/witchcraft-go-error"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts from v1alpha1 to the storage version v1alpha2
+func (d *Demand) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha2.Demand)
```

Comment:
You shouldn't panic if this is not the case. Return an error instead.

Reply:
From what I looked at the conversion code this should be a safe conversion. However, will still add the errors just in case